### DISCUSSION TITLE
Can someone help me understand plugins?

### DISCUSSION THREAD

**Discussion by vvevans** [nornir-automation/nornir#659] [Q&A]
I'm new to Python and new to Nornir. I'm trying to understand registering a plugin (specifically, netmiko) for use. The nornir docs are vague and I'm trying to understand this "setup.py" file, poetry, etc.

Can anyone help me understand this or point me in the direction of a HOWTO?

**Comment by dbarrosop** [comment]
You only need to register them if you are developing those plugins. If you are leveraging already made ones that part is most likely being taken care of for you. So if you are using the official netmiko plugins you don't need to even think about this.

Otherwise, you have info here:

https://nornir.readthedocs.io/en/latest/plugins/index.html
